src/App.tsx

```diff
@@ -1,51 +1,80 @@
+import goodsFromServer from './goods';
 
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
+type State = {
+  goods: string[],
+  selectedGood: string,
+};
 
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
+export class App extends Component<{}, State> {
+  state = {
+    goods: [],
+    selectedGood: goodsFromServer[8],
+  };
 
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
+  componentDidMount() {
+    this.setState({ goods: goodsFromServer });
+  }
 
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
+  render() {
+    const { goods, selectedGood } = this.state;
 
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
-};
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+            {selectedGood.length > 0 ? `${selectedGood} is selected` : 'No goods selected'}
+          </h1>
+          {selectedGood.length > 0
+          && (
+            <button
+              type="button"
+              className="App__clear"
+              onClick={() => {
+                this.setState({ selectedGood: '' });
+              }}
+            >
+              Clear
+            </button>
+          )}
+        </header>
+
+        <ul>
+          {goods.map(good => (
+            <li
+              className={good === selectedGood ? 'Good Good--active' : 'Good'}
+            >
+              {selectedGood !== good
+                && (
+                  <button
+                    type="button"
+                    className="Good__select"
+                    onClick={() => {
+                      this.setState({ selectedGood: good });
+                    }}
+                  >
+                    Select
+                  </button>
+                )}
+              {selectedGood === good
+                && (
+                  <button
+                    type="button"
+                    className="Good__remove"
+                    onClick={() => {
+                      this.setState({ selectedGood: '' });
+                    }}
+                  >
+                    Remove
+                  </button>
```

Comment:
`selectedGood !== good` you use this expression several times, makes sense to create a separate variable for this.